File: designPattern/Observer/python/Generator.py
import abc


class Generator:
    __metaclass__ = abc.ABCMeta

    def __init__(self):
        self.observers = []

    def add_observer(self, observer):
        self.observers.append(observer)

    def allNotify(self):
        it = iter(self.observers)
        for i in range(len(self.observers)):
            observer = next(it)
            observer.update(self)

    @abc.abstractmethod
    def execute(self):
        pass

    @abc.abstractmethod
    def getSon(self):
        pass

    @abc.abstractmethod
    def getDaughter(self):
        pass

File: designPattern/Observer/review/Generator.java
import java.util.Iterator;
import java.util.ArrayList;

public abstract class Generator{
	ArrayList observers = new ArrayList();
	public void addObserver(Observer observer){
		observers.add(observer);
	}
	public void allNotify(){
		Iterator it = observers.iterator();
		while(it.hasNext()){
			Observer observer = (Observer)it.next();
			observer.update(getNumber());
		}
	}
	public abstract int getNumber();
	public abstract void generate();
}